upgrader/apps/upgrader/lib/filter/ActionFilter.php

```diff
@@ -3,8 +3,14 @@
 class ActionFilter extends sfFilter {
 
     public function execute($filterChain) {
+        $rootPath = realpath(dirname(__FILE__) . '/../../../../..');
 
         $request = $this->getContext()->getRequest();
+        if (is_file($rootPath . '/lib/confs/Conf.php')) {
+            header("Location: /index.php");
+            exit();
```

Comment:
Yes its good to call exit after  header redirection 
in case of some one try to access using CURL  with out -L option